File: app/src/main/java/com/telecorp/dashqueue/api/model/HospitalItem.java
package com.telecorp.dashqueue.api.model;

import com.google.gson.annotations.SerializedName;

/**
 * Created by Saran on 13/10/2560.
 */

public class HospitalItem {

    @SerializedName("UID")
    private long uid;
    @SerializedName("code")
    private String code;
    @SerializedName("title")
    private String title;
    @SerializedName("active")
    private String active;
    @SerializedName("urlapl")
    private String urlapl;
    @SerializedName("ImageLogo")
    private String imageLogo;
    @SerializedName("PathImageLogo")
    private String imageLogoPath;

    public HospitalItem() {
    }

    private HospitalItem(long uid, String code, String title, String active, String urlapl, String imageLogo, String imageLogoPath) {
        this.uid = uid;
        this.code = code;
        this.title = title;
        this.active = active;
        this.urlapl = urlapl;
        this.imageLogo = imageLogo;
        this.imageLogoPath = imageLogoPath;
    }

    public long getUid() {
        return uid;
    }

    public String getCode() {
        return code;
    }

    public String getTitle() {
        return title;
    }

    public String getActive() {
        return active;
    }

    public String getUrlapl() {
        return urlapl;
    }

    public String getImageLogo() {
        return imageLogo;
    }

    public String getImageLogoPath() {
        return imageLogoPath;
    }


    public static HospitalItem getMock(String hello){
        return new HospitalItem(1L,hello+"1",hello,hello,hello,hello,hello);
    }
}

File: app/src/main/java/com/telecorp/dashqueue/api/model/HospitalItem.kt
package com.telecorp.dashqueue.api.model

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import com.page365.store365.generic.searchable.SearchableEntity

/**
 * Created by Saran on 13/10/2560.
 */

class HospitalItem() : SearchableEntity, Parcelable {
    @SerializedName("UID")
    var uid: Long? = null

    @SerializedName("code")
    var code: String? = null

    @SerializedName("title")
    var title: String? = null

    @SerializedName("active")
    var active: String? = null

    @SerializedName("urlapl")
    var urlapl: String? = null

    @SerializedName("ImageLogo")
    var imageLogo: String? = null

    @SerializedName("PathImageLogo")
    var imageLogoPath: String? = null

    constructor(parcel: Parcel) : this() {
        uid = parcel.readValue(Long::class.java.classLoader) as? Long
        code = parcel.readString()
        title = parcel.readString()
        active = parcel.readString()
        urlapl = parcel.readString()
        imageLogo = parcel.readString()
        imageLogoPath = parcel.readString()
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeValue(uid)
        parcel.writeString(code)
        parcel.writeString(title)
        parcel.writeString(active)
        parcel.writeString(urlapl)
        parcel.writeString(imageLogo)
        parcel.writeString(imageLogoPath)
    }

    override fun getCriteria(): String {
        return title.toString();
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<HospitalItem> {
        override fun createFromParcel(parcel: Parcel): HospitalItem {
            return HospitalItem(parcel)
        }

        override fun newArray(size: Int): Array<HospitalItem?> {
            return arrayOfNulls(size)
        }
    }


}
